upload-chunked-file.js

```diff
@@ -0,0 +1,43 @@
+function chunkFileUpload(file, serverUrl, chunkSize) {
```

Comment:
The function should return a Promise or use async/await to handle asynchronous operations cleanly, instead of relying on nested callbacks.